### ISSUE TITLE
Enhance matchers with ability to skip fields when comparing objects/collections/cursors

### ISSUE THREAD

**Issue by jgebal** [utPLSQL/utPLSQL#183]
## Background
When comparing two objects, collections or cursors I'd like to be able to provide a list of field names that I don't want to compare.
This is very useful, when object/record contain fields like `create_date`, `last_modified_date`.
The matcher should allow users, to narrow down the scope of comparison for complex types.

## Expected call syntax
When building expectation the call could look like this:
```sql
ut.expect( a_data_cursor ).to_equal(
  a_expected_cursor, a_skip_fields => ignore_fields('create_date') 
);
```
The `ignore_fields` would be just a synonym to `ut_varchar2_list`
The argument `a_skip_fields` should be optional and be null by default.
The argument `a_skip_fields` should be before `a_nulls_are_equal` on the argument list, so that the below syntax is also valid.
```sql
ut.expect( a_data_cursor ).to_equal( a_expected_cursor, ignore_fields('create_date') );
```

**Comment by jgebal** [comment]
After a discussion today at work I realised that we might need to have two optional parameters:`a_include => ...`, `a_exclude => ...`
The list of columns/attributes that are getting compared should be filtered by both, so that:
If query has columns:
`a, b, c, d, e, f`
and we specify parameters: `a_include => strings('a','b','d'), a_exclude => strings('b')`
then only fields/attributes 'a' and 'd' get compared and all the rest is skipped.

**Comment by jgebal** [comment]
Exclusions resolved by #405
